stubs/regex/regex/_regex.pyi

```diff
@@ -0,0 +1,185 @@
+from _typeshed import Self
+from typing import Any, AnyStr, Callable, Generic, Mapping, TypeVar, overload
+
+_T = TypeVar("_T")
+
+class Pattern(Generic[AnyStr]):
+
+    pattern: AnyStr
+    flags: int
+    groups: int
+    groupindex: Mapping[str, int]
+    named_lists: Mapping[str, frozenset[AnyStr]]
+    def search(
+        self,
+        string: AnyStr,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> Match[AnyStr] | None: ...
+    def match(
+        self,
+        string: AnyStr,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> Match[AnyStr] | None: ...
+    def fullmatch(
+        self,
+        string: AnyStr,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> Match[AnyStr] | None: ...
+    def split(
+        self, string: AnyStr, maxsplit: int = ..., concurrent: bool | None = ..., timeout: int | None = ...
+    ) -> list[AnyStr | Any]: ...
```

Comment:
The method in `re` returns `list[AnyStr]`. Is there a reason for this to differ?

Reply:
Ah yes this is an error in `re` I've added the fix in #6762 😄